packages/rest-contracts-lambda/src/rest-contracts-lambda.ts

```diff
@@ -0,0 +1,191 @@
+ * the first (event) parameter is a copy for which:
+ *   (1) the type for pathParameters is RestContracts.PATH_PARAMS<typeof api>
+ *   (2) the type for queryStringParameters is RestContracts.QUERY_PARAMS<typeof api>
+ *   (3) a new "bodyObject" attribute contains the JSON-parsed body
+ *       (if any) of type RestContracts.BODY_PARAMS<typeof api>
+ *   (4) a new "params" attribute combines the query parameters, path parameters, and
+ *       (for POST methods only) the body parameters.  Body parameters are kept separate
+ *       for PUT/PATCH methods because the body typically represents an object to be stored
+ *       (PUT) or fields of that object (PATCH) and not individual parameters.
+ *       If parameters in the body, query, or path have the same name, those in the query
+ *       parameter take precedence (will be written over) those in the body parameter
+ *       (again, body only being included for POST),
+ *       and those in the path parameter will overwrite those in the query/body parameter.
+ *       (This ordering was chosen because attackers are more likely to have control
+ *        of the body contents than the query, and more likely to have control over
+ *        the query parameters than the path parameters.)
+ * @param errorHandler An optional error handler that catches exceptions and
+ * handles them by calling the standard lambda callback function.  If not set,
+ * a default exception handler looks for the HTTP status to return in either the
+ * statusCode or status attribute of the error that was thrown, and sets the
+ * response body to the JSON encoding of the error.  If neither statusCode nor
+ * status is set to a number, the default handler calls callback() with the
+ * error as the first parameter and lets lambda figure out how to handle it
+ * based on your configured integration reponse, as documented in:
+ * https://docs.aws.amazon.com/apigateway/latest/developerguide/handle-errors-in-lambda-integration.html
+ */
+export const wrapLambdaParametersAndResult = <API extends RestContracts.API>(
+  api: API,
+  handler: (
+    event: AugmentedAPIGatewayProxyEvent<API>,
+    context: Lambda.Context
+  ) => PromiseLike<RestContracts.RESULT<API>> | RestContracts.RESULT<API>,
+  errorHandler: (
+    err: any,
+    callback: Lambda.APIGatewayProxyCallback
+  ) => void = defaultErrorHandler
+) => wrapLambdaParameters( api, async (
+  event: AugmentedAPIGatewayProxyEvent<API>,
+  context: Lambda.Context,
+  callback: Lambda.APIGatewayProxyCallback
+) => {
+  try {
+    const result = await handler( event, context );
+    callback(undefined, {
+      // tslint:disable:no-magic-numbers
+      statusCode:
+        (api.method === RestContracts.Method.get && typeof(result) === "undefined") ? 404 :
+        (api.method === RestContracts.Method.put) ? 201 :
+        200,
+      // tslint:enable:no-magic-numbers
+      body: JSON.stringify(result)
```

Comment:
Could we ensure that result is JSON string?

Reply:
I think you are asking to allow multiple types of result types: not just JSON, but raw strings or possibly even binary that would be base64 encoded?  I just checked in a change that will do just that.

While I was add it, I added CORS support.